client/src/components/RegisterUserModal.jsx

```diff
@@ -0,0 +1,166 @@
+import {
+  useDisclosure,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+  Input,
+  Select,
+  useToast,
+} from '@chakra-ui/react';
+import { StyledButton } from './views/Login.jsx';
+import { Text } from './common/Typography.jsx';
+import React, { Fragment, useRef, useState } from 'react';
+import styled from '@emotion/styled';
+import { makeRequest } from '../utils/api.js';
+
+const ValidationText = styled(Text)`
+  font-size: 10px;
+  padding-left: 12px;
+`;
+
+const rolesAvailable = ['Admin', 'General'];
+
+const RegisterUserModal = () => {
+  const usernameRef = useRef('');
+  const passwordRef = useRef('');
+  const emailRef = useRef('');
+  const roleRef = useRef('');
+  const [passwordIsValidated, setPasswordIsValidated] = useState(false);
+  const [emailIsValidated, setEmailIsValidated] = useState(false);
+  const [buttonIsDisabled, setButtonIsDisabled] = useState(false);
+  const toast = useToast();
+  const [isLoading, setIsLoading] = useState(false);
+  const { isOpen, onOpen, onClose } = useDisclosure();
+
+  const registerHandleSubmit = async (e) => {
+    e.preventDefault();
+    setIsLoading(true);
+
+    try {
+      await makeRequest('post', 'user/register', {
+        email: emailRef.current.value,
+        password: passwordRef.current.value,
+        role: roleRef.current.value,
+        username: usernameRef.current.value,
+      });
+
+      toast({
+        title: 'Request Sent',
+        description:
+          'The request has been sent, you will receive an email when the account has been approved',
+        status: 'success',
+        duration: 9000,
+        isClosable: true,
+        onCloseComplete: onClose,
+      });
+    } catch {
+      toast({
+        title: 'Error',
+        description: 'Email or username already exists',
```

Comment:
should we just show the error response from the server here?

Reply:
We tried it and it only returns "Bad request status 400" or some shit.

I think it's the makeRequest is abstracting too much so I'll refactor that with Sujan on the next ticket.

Reply:
its response.data that you need

Reply:
but ya no worries

Reply:
> its response.data that you need

Oh yeah you're right oops

Reply:
or like error.response.data or something like that

Reply:
![image](https://user-images.githubusercontent.com/4203037/108266085-a7f62b00-7137-11eb-97e2-810b137380b4.png)

Is this good?

Reply:
I feel like it gives too much information to the user, but at the same time it's not a *real* app

Reply:
nah lol I need to fix the error validation for mongoose. Its ok keep the generic error message for now